### PR TITLE
change labels() method return type

### LINKED ISSUE #57: `labels()` returns `_Metric` type instead of `Self`
```py
from pytheus.metrics import Gauge

gauge = Gauge(
    "room_temperature", "temperature in the living room", ["room_name"]
)

gauge.labels(room_name="bedroom").set(25)

# pyright throws an error
# Cannot access attribute "set" for class "_Metric"
#
# gauge.labels(room_name="bedroom") should return Gauge type, not _Metric
```

### PR DESCRIPTION
**Pull request by pradishb**
fix #57

### CODE REVIEW

File: pytheus/metrics.py [hunk 1]
```diff
@@ -16,6 +27,10 @@
 from pytheus.registry import REGISTRY, Collector, Registry
 from pytheus.utils import InfFloat, MetricType
 
+# Backwards compatibility for python<3.11
+if TYPE_CHECKING:
+    from typing import Self
```
Comment (Llandy3d) [inline pytheus/metrics.py:32]: I would skip this check and import directly a backward compatible backport 🤔

File: pytheus/metrics.py [hunk 2]
```diff
@@ -202,7 +217,7 @@ def _raise_if_cannot_observe(self) -> None:
         if not self._can_observe:
             raise UnobservableMetricException
 
-    def labels(self, pytheus_labels_: Optional[Labels] = None, **kwargs: str) -> "_Metric":
+    def labels(self, pytheus_labels_: Optional[Labels] = None, **kwargs: str) -> "Self":
```
Comment (Llandy3d) [inline pytheus/metrics.py:220]: If the suggestion from this review gets applied, I would remove the quotes and use directly `Self`

**Review by Llandy3d — CHANGES_REQUESTED** [review]
👋  hi, thanks a lot for catching this issue and the contribution!

Due to having to support older versions of python I think the best course of action would be to use the [typing_extensions](https://github.com/python/typing_extensions) library to make direct use of `Self`.

From my tests using `"Self"` would still mark it as `unknown` for pyright the returned object (although it fixes the error on the method).

Let me know if you want to work on it further or if you prefer for me to take over, regardless of choice I really appreciate you taking the time for raising and fixing this 🙌 

So to recap the course of action would be:
1. Add a dependency to typing_extensions
2. Import `Self` without the need for the `TYP_CHECKING` check
3. Replace `"Self"` -> `Self`

**Comment by pradishb** [comment]
I agree that using typing_extensions is the better solution. I have made the requested changes,

**Review by Llandy3d — APPROVED** [review]
Looks good! 🙌 
Thank you for your contribution 🙇

**Comment by Llandy3d** [comment]
it actually doesn't seem to be enough as the return type now mismatch 😭

File: pytheus/metrics.py [hunk 3]
```diff
@@ -146,7 +163,7 @@ def __init__(
         labels: Optional[Labels] = None,
         default_labels: Optional[Labels] = None,
         registry: Optional[Registry] = REGISTRY,
-        collector: Optional[_MetricCollector] = None,
+        collector: Optional[_MetricCollector[Self]] = None,
```
Comment (Llandy3d) [inline pytheus/metrics.py:166]: this needs to be changed also for the `Histogram` and `Summary` class 🤔

**Review by Llandy3d — COMMENTED** [review]
Looks good!

I think that for completeness we should also align the typing for the mentioned classes, what do you think ?
Beside that looks great 🙌
